src/readthedocs-config.js

```diff
@@ -59,63 +61,80 @@ export function getReadTheDocsConfig(sendUrlParam) {
+
+    fetch(defaultApiUrl, {
+      method: "GET",
     })
-    .then((data) => {
-      // We force the user to define the `<meta>` tag to be able to use Read the Docs data directly.
-      // This is to keep forward/backward compatibility without breaking integrations.
-      const metadataAddonsAPIVersion = _getMetadataAddonsAPIVersion();
-      if (metadataAddonsAPIVersion !== undefined) {
-        if (metadataAddonsAPIVersion !== data.api_version) {
-          // When the API scheme version returned doesn't match the one defined via `<meta>` tag by the user,
-          // we perform another request to get the Read the Docs response in the structure
-          // that's supported by the user and dispatch a custom event letting them know
-          // this data is ready to be consumed under `event.detail`.
-
-          url =
-            ADDONS_API_ENDPOINT +
-            new URLSearchParams({
-              url: window.location.href,
-              "client-version": CLIENT_VERSION,
-              "api-version": metadataAddonsAPIVersion,
-            });
+      .then((response) => {
+        if (!response.ok) {
+          reject("Error hitting addons API endpoint");
+        }
+        // Use the addons API data response as `dataUser`
+        dataUser = response.json();
+        return dataUser;
+      })
+      .then((data) => {
+        // Create a new Promise here to handle the user request in a different async task.
+        // This allows us to start executing our integration independently from the use one.
+        new Promise((resolve, reject) => {
+          // We force the user to define the `<meta>` tag to be able to use Read the Docs data directly.
+          // This is to keep forward/backward compatibility without breaking integrations.
+          const metadataAddonsAPIVersion = _getMetadataAddonsAPIVersion();
 
-          fetch(url, {
-            method: "GET",
-          })
-            .then((response) => {
+          if (
+            metadataAddonsAPIVersion !== undefined &&
+            metadataAddonsAPIVersion !== data.api_version
+          ) {
+            // When the API scheme version returned doesn't match the one defined via `<meta>` tag by the user,
+            // we perform another request to get the Read the Docs response in the structure
+            // that's supported by the user and dispatch a custom event letting them know
+            // this data is ready to be consumed under `event.detail`.
+            const userApiUrl =
+              ADDONS_API_ENDPOINT +
+              new URLSearchParams({
+                url: window.location.href,
+                "client-version": CLIENT_VERSION,
+                "api-version": metadataAddonsAPIVersion,
+              });
+
+            fetch(userApiUrl, {
+              method: "GET",
+            }).then((response) => {
               if (!response.ok) {
-                throw "Error parsing configuration data";
+                reject(
```

Comment:
One thing to be careful of with using `Promise` is to make sure you are using `return` to end the function on resolve/reject, otherwise this will continue on in the promise chain.

```suggestion
                return reject(
```